File: 2021/day1/part1/go/sonar.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {

	file, err := os.ReadFile("../../input.txt")

	if err != nil {
		panic(err)
	}

	s := string(file)

	vals := strings.Fields(s)

	count := 0

	prev, err := strconv.Atoi(vals[0])

	if err != nil {
		panic(err)
	}

	for _, val := range vals {

		n, err := strconv.Atoi(val)

		if err != nil {
			panic(err)
		}

		if prev < n {
			prev = n
			count++
		} else if prev > n {
			prev = n
		}

	}

	fmt.Println(count)

}

File: 2021/day1/part1/js/Sonar.js
const fs = require("fs")



fs.readFile("../../input.txt", { encoding: 'utf8' }, function callback(err, data) {

    const arr = data.trim().split("\n")

    let prev = Number(arr[0])
    let count = 0

    for (let i of arr) {
        let n = Number(i)

        if (prev < n) {
            console.log(n + " (increased)")
            prev = n
            count++
        } else if (prev > n) {
            console.log(n + " (decreased)")
            prev = n
        } else {
            console.log(n + " (N/A - no previous measurement)")
        }

    }

    console.log(count)
})
